File: android/src/main/kotlin/com/ovaom/flutter_libpd/pd_controller/pd_patch_file_manager/PdPatchFileManager.kt
package com.ovaom.flutter_libpd.pd_controller.pd_patch_file_manager

interface PdPatchFileManager {
    val openedPatches: List<PatchFile>
    fun openPatch(file: PatchFile)
    fun closePatch(file: PatchFile)
}

data class PatchFile(val name: String, val path: String) {
    val fullPath: String
    get() = "$path$name"
}

sealed class PatchFileError : Exception() {
    data class OpenFailed(val file: PatchFile): PatchFileError()
}

File: ios/Classes/PdController/PdPatchFileManager/PdPatchFileManager.swift
//
//  PatchFileManager.swift
//  Runner
//
//  Created by flav on 24/05/2022.
//

import Foundation

public protocol PdPatchFileManager {
    var openedPatches: [PatchFile] { get }
    func openPatch(_ file: PatchFile) throws
    func closePatch(_ file: PatchFile)
}

public struct PatchFile: Equatable {
    public let name: String
    public let path: String

    public init(name: String, path: String) {
        self.name = name
        self.path = path
    }
    
    public var fullPath: String { path + name }
}

public enum PatchFileError: Error {
    case openFailed(PatchFile)
}
